synapse/models/inet.py

```diff
@@ -602,31 +249,404 @@ def norm(self, valu, oldval=None):
             subs['port'] = port
 
         if resauth:
-            hostpart = '%s@%s' % (resauth,hostpart)
+            hostpart = '%s@%s' % (resauth, hostpart)
 
-        valu = '%s://%s/%s' % (proto,hostpart,respath)
-        return valu,subs
+        valu = '%s://%s/%s' % (proto, hostpart, respath)
+        return valu, subs
 
     def repr(self, valu):
         return valu
 
-
 class CidrType(DataType):
-
     def norm(self, valu, oldval=None):
-
-        ipstr,maskstr = valu.split('/')
+        ipstr, maskstr = valu.split('/')
 
         mask = int(maskstr)
         ipv4 = ipv4int(ipstr)
 
         if mask > 32 or mask < 0:
             self._raiseBadValu(valu, mesg='Invalid CIDR Mask')
 
-        ipv4 = ipv4mask(ipv4,mask)
-        valu = '%s/%d' % ( ipv4str(ipv4), mask )
+        ipv4 = ipv4mask(ipv4, mask)
+        valu = '%s/%d' % (ipv4str(ipv4), mask)
 
-        return valu,{'ipv4':ipv4,'mask':mask}
+        return valu, {'ipv4': ipv4, 'mask': mask}
 
     def repr(self, valu):
         return valu
+
+class InetMod(CoreModule):
+
+    def initCoreModule(self):
+        # add an inet:defang cast to swap [.] to .
+        self.core.addTypeCast('inet:defang', castInetDeFang)
+        self.onFormNode('inet:fqdn', self.onTufoFormFqdn)
+        self.onFormNode('inet:passwd', self.onTufoFormPasswd)
```

Comment:
It was useful since it provided a very clean way to just say "on this node-type creation, do X as well".  From a code-reading perspective, these helpers provide a easy way to signal that "I register these actions when I get loaded".

We should plumb helpers for additional actions (node:set comes to mind) but I didn't want to plumb those changes here in order to keep the delta focused.